### PR TITLE
Fix Backward loop issue on pytorch 2.5

### PR DESCRIPTION
**Pull request by ybh1998**
When I tried to back propagate on PyTorch 2.5.1, the program stuck at `loss.backward()`. After some investigation, I found the following issue:

When I add a breakpoint at https://github.com/aliutkus/torchinterp1d/blob/4e7533dce96edf4aa4549bc75cf532d9e782befb/torchinterp1d/interp1d.py#L155

The `grad_fn` of `ctx.saved_tensors[0]` is `<Interp1dBackward>`, causing a loop backward call.

When I add a breakpoint at https://github.com/aliutkus/torchinterp1d/blob/4e7533dce96edf4aa4549bc75cf532d9e782befb/torchinterp1d/interp1d.py#L149

The `grad_fn` of `ynew` is  `<AddBackward0>`, which means the grad_fn of `ctx.saved_tensors[0]` is touched between forward and backward. I tried to add `.detach()` after calling `interp1d`, but the results were the same.

I suspect that the PyTorch auto_grad engine modifies the `grad_fn` of `ynew` in-place after some version. Adding a `detach()` in the forward function resolves this problem.

### CODE REVIEW

**Comment by Nikhil0504** [comment]
@aliutkus could you please push this since this was the error I was finding as well!
